driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -278,48 +279,64 @@ void sendRequest() {
+                    } catch (ConnectionException e) {
+                        // If we have any problem with the connection, move to the next node.
+                        if (metricsEnabled())
+                            metrics().getErrorMetrics().getConnectionErrors().inc();
+                        if (connection != null)
+                            connection.release();
+                        logError(host.getSocketAddress(), e);
+                        findNextHostAndQuery();
+                    } catch (BusyConnectionException e) {
+                        // The pool shouldn't have give us a busy connection unless we've maxed up the pool, so move on to the next host.
+                        connection.release();
+                        logError(host.getSocketAddress(), e);
+                        findNextHostAndQuery();
+                    } catch (RuntimeException e) {
+                        if (connection != null)
+                            connection.release();
+                        logger.error("Unexpected error while querying " + host.getAddress(), e);
+                        logError(host.getSocketAddress(), e);
+                        findNextHostAndQuery();
+                    }
                 }
-                current = host;
-                write(connection, this);
-                return true;
-            } catch (ConnectionException e) {
-                // If we have any problem with the connection, move to the next node.
-                if (metricsEnabled())
-                    metrics().getErrorMetrics().getConnectionErrors().inc();
-                if (connection != null)
-                    connection.release();
-                logError(host.getSocketAddress(), e);
-                return false;
-            } catch (BusyConnectionException e) {
-                // The pool shouldn't have give us a busy connection unless we've maxed up the pool, so move on to the next host.
-                connection.release();
-                logError(host.getSocketAddress(), e);
-                return false;
-            } catch (TimeoutException e) {
-                // We timeout, log it but move to the next node.
-                logError(host.getSocketAddress(), new DriverException("Timeout while trying to acquire available connection (you may want to increase the driver number of per-host connections)", e));
-                return false;
-            } catch (RuntimeException e) {
-                if (connection != null)
-                    connection.release();
-                logger.error("Unexpected error while querying " + host.getAddress(), e);
-                logError(host.getSocketAddress(), e);
-                return false;
-            }
+
+                @Override
+                public void onFailure(Throwable t) {
+                    if (t instanceof TimeoutException) {
+                        logError(host.getSocketAddress(),
```

Comment:
Is `TimeoutException` possible?  I noticed the `Pool is Busy` exception is a `ConnectionException`.